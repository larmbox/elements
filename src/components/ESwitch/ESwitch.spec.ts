import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { ESwitch } from '~/index';

test('mount component', async () => {
  expect(ESwitch).toBeTruthy();

  const component = mount(ESwitch, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain(
    'EFormControl ESwitch ESwitch--md ESwitch--primary'
  );
  expect(component.html()).not.toContain('form-field');
});

test('register css variables', async () => {
  mount(ESwitch);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('ESwitch');
});

test('inherit attributes', async () => {
  const component = mount(ESwitch, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
