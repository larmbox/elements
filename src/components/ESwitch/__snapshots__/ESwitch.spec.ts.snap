// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"EFormControl ESwitch ESwitch--md ESwitch--primary\\">
  <div class=\\"ESwitch-inner\\">
    <div><input id=\\"id\\" type=\\"checkbox\\" />
      <div class=\\"ESwitch-box\\">
        <div class=\\"ESwitch-inner-box\\"><span class=\\"EIcon check ESwitch-icon\\" aria-hidden=\\"true\\"></span><span class=\\"EIcon x ESwitch-icon\\" aria-hidden=\\"true\\"></span></div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;
